src/couch/src/couch_db.erl

```diff
@@ -914,7 +914,18 @@ prep_and_validate_update(Db, #doc{id=Id,revs={RevStart, Revs}}=Doc,
         % new doc, and we have existing revs.
         % reuse existing deleted doc
         if OldFullDocInfo#full_doc_info.deleted orelse AllowConflict ->
-            {validate_doc_update(Db, Doc, fun() -> nil end), Doc};
+            case couch_doc:has_stubs(Doc) of
+                true ->
+                    #full_doc_info{rev_tree=RevTree} = OldFullDocInfo,
+                    Leafs = couch_key_tree:get_all_leafs(RevTree),
+                    [{Leaf, DiskRevs} | _] = lists:reverse(Leafs),
```

Comment:
I can't convince myself that this is correct. Is there some sort of guarantee that we're getting on the order of leaves returned by `get_all_leafs/1`? If not it seems like we need be looking at adding a sort based on `{Depth, Rev}` for the returned leaves. Though that's also assuming that we're only allowing the un-deletion of the latest rev. 

The test here would be when the deleted doc has multiple deleted conflicts of which only some have the required missing attachments. Given that the new doc comes in without a revision it seems like we'd need to pick a deterministic one? Though that confuses me on when a different deleted leaf revision would satisfy the attachment conditions...

Reply:
@eiri two questions:

1) Will this be ready in the next few days?

2) Are you subscribed to dev@couchdb.apache.org and seeing my regular emails about the 2.3.0 milestone?

Reply:
1. Maybe, but I can't guarantee that. I just closed another blocker yesterday and working on this now, but here is also inclusion of changes for `active_size` we've discussed on dev mail list which makes a scope of this PR bigger.
2. Yes I am. Sorry, I'm working on those issues as fast as I can.

Reply:
@eiri thanks :) no problem. If this is going to be too big to get into 2.3.0, that's fine -- and I know people in the US are off until Monday -- so willing to be lenient here if this is still making progress.

Thank you again for all your hard work :)

Reply:
@wohali I'm afraid I can't make it to 2.3.0 😞 I've spend half of the weekend on this, but `active` still not calculated as expected in my tests and I yet to find what I'm missing there. I'll continue to work on this, but I don't see myself wrapping it up today.

Reply:
@eiri that's fine, i'd rather this be correct than rushed. Thanks for the heads up!